telemetry/src/test/java/com/newrelic/telemetry/TelemetryClientTest.java

```diff
@@ -219,12 +252,18 @@ void retryWithSplitGoesOverLimit() throws Exception {
 
     TelemetryClient testClass =
         new TelemetryClient(batchSender, null, null, null, 1, true, maxTelemetry);
-
+    CustomNotificationHandler customNotificationHandler = new CustomNotificationHandler();
+    testClass.withNotificationHandler(customNotificationHandler);
     testClass.sendBatch(batch);
     testClass.shutdown();
     boolean result = sendLatch.await(3, TimeUnit.SECONDS);
     assertTrue(result);
     assertEquals(1, sendCount.get());
+    assertEquals(0, customNotificationHandler.errorMessages.size());
+    assertEquals(1, customNotificationHandler.infoMessages.size());
+    assertEquals(
+        "Metric batch size too large, splitting and retrying.",
+        customNotificationHandler.infoMessages.get(0));
```

Comment:
Consider combining these types of assertions:
```
    assertEquals(Collections.emptyList(), customNotificationHandler.errorMessages);
    assertEquals(Collections.singletonList("Metric batch sending failed. Retrying failed batch after 15 MILLISECONDS"),
            customNotificationHandler.infoMessages);
```
(This looks nicer with Java 11's `List.of()` - we should think about using Java 11+ compatibility for test code)